### ISSUE TITLE
Watcher ignores Picker and Encoding settings

### ISSUE THREAD

**Issue by snottyone** [RandomEngy/VidCoder#1097]
### Problem Description

two folders watched, with different picker and encoding settings for each.
files in both folders, enable watcher.
all files get queued, but with the current VidCoder picker & encoding settings, not the ones specified in Watcher

### What version of VidCoder are you running?

8.14 Beta SquirrelInstaller

**Comment by RandomEngy** [comment]
Thanks for the report. This is fixed in 8.15 Beta.

**Comment by RandomEngy** [comment]
To be a bit more specific I fixed an issue where the encode directory wasn't picked up correctly from the configured picker. Let me know if you see any other settings that are not being honored.

**Comment by snottyone** [comment]
beta 8.15: watcher still picks the current encoding preset over the one set in watcher. it got the picker right though.

**Comment by RandomEngy** [comment]
Found another issue where it doesn't pick the correct file extension (.mp4 vs .mkv, for example). What did you see that was different about the encoding preset?

**Comment by snottyone** [comment]
i've not seen problems with extensions, watcher gets them right for me (i encode two different sorts of mp4, one to mkv, one to mp4. via picker settings. that works.

it also gets the picker partly wrong.
latest test: it got the picker's encode directory right, but not that picker's naming scheme.
it still grabs whatever encoding setting is set in the main app and runs with that.

**Comment by snottyone** [comment]
one picker's naming scheme is {parent}\{source} {preset}
watcher gets the filename part right, but not the parent folder

**Comment by RandomEngy** [comment]
I tested changing the video encoder and it gets that correct.

But I did confirm that the picker's naming scheme is not correctly picked up That one should be fixed in 8.16 Beta.

**Comment by snottyone** [comment]
could you maybe reopen this? it's hard to find each time....

i just tested with 8.16b and get correct picker results afaics. the paths are correct, including the {parent} tag
it still grabs the current encoding setting (which was neither of the two desired ones for this test)

a request: it would be much easier to work with the watcher window if the configured watch folder were visible and editable when watcher is disabled.

**Comment by RandomEngy** [comment]
Hmm, the encoding settings are applied fine for me; at least in checking the video encoder. If you send me your %appdata%\VidCoder-Beta\VidCoder.sqlite file and a series of steps to reproduce on your side I'll take another look.

**Comment by snottyone** [comment]
the new non-beta version still has the same bug. Watcher encoding settings are ignored and taken from what is set in VidCoder.

**Comment by snottyone** [comment]
and there is a NEW bug!
it adds newly arrived files in a watched folder twice (with the wrong settings of course)

**Comment by snottyone** [comment]
video 
https://user-images.githubusercontent.com/20132137/212549716-37031cce-da46-4905-a38f-0adece51930e.mp4

**Comment by RandomEngy** [comment]
I loaded up your settings, changed the current settings to x264 and the output video was still correctly as H.265 as per the settings specified in the watcher. This is why I wanted to know *specifically* what steps you took, what particular setting you observed was incorrect and what you expected to see. Perhaps you were checking a specific filter or audio setting instead?

As for the double-encode, could you check to see how many instances of VidCoderFileWatcher.exe are running? This tool can easily search by process name: https://learn.microsoft.com/en-us/sysinternals/downloads/process-explorer . Also, what install method did you use? Do you have any other instances of VidCoder on your computer? I'm wondering if maybe another install with different config settings is running in the background and queuing the erroneous jobs.

**Comment by snottyone** [comment]
i have the stable and the beta installed, i will remove the beta, now that stable has the watcher and try again.
fwiw, 8.20 just did the same thing, picking the current encoding setting over the one set in watcher.

**Comment by snottyone** [comment]
uninstalled beta version. rebooted tbs.
made a new watched folder entry - same thing, picker setting is ok, encode setting is taken from what is set in the main window.

**Comment by snottyone** [comment]
not opening another bug:
watcher does not scan folders any more. i put new files in watched folder, nothing happening.
disable/enable watch folders, new files are found

**Comment by RandomEngy** [comment]
The video did not tell me what encode setting you were checking to see if it was picked up correctly.

Also please let me know if you see multiple instances of VidCoderFileWatcher.exe .

And in %appdata%\VidCoder\Logs you can find files that end in Watcher.txt that should log when they send a file to be encoded. For the duplicate and the folders it failed to scan, could you send those log files over? It might be that the watcher crashed or had some other problem and logged it there.

**Comment by snottyone** [comment]
![2023-01-15 - 16-14-00 VidCoder mp4 - 00;00;15 673](https://user-images.githubusercontent.com/20132137/212806461-8086516a-8033-48fe-9e0a-88c2c6cfac2a.jpg)

**Comment by snottyone** [comment]
[2023-01-15 15.53.48 Watcher.txt](https://github.com/RandomEngy/VidCoder/files/10431203/2023-01-15.15.53.48.Watcher.txt)

that's the watcher file for the encode in the video

**Comment by RandomEngy** [comment]
Thanks. The "wrong settings" issue is just an issue with the displayed preset name in the queue. It's actually applying the correct settings from the preset specified in the watcher. I'll fix the display issue in the next release; thanks for reporting it.

As for the double encoding, I can see from the log that it's not from a separate process. And that the watched folder list isn't duplicated. From the logs I can also see that the folder watcher was set up only once, but on your system the FileSystemWatcher is triggering every event twice. I'll see what I can do to work around this. But if the file watcher API is acting up that could also explain why it's missing added files in some cases. Though to be sure if you get that issue again where the watcher is not finding files, I'd love a Watcher.txt log for that.

**Comment by RandomEngy** [comment]
Okay, I released v8.21, which should fix the preset display name issue and also includes an attempted fix for double-firing FileSystemWatcher events, which your system seems to have. Let me know if this helps.

**Comment by snottyone** [comment]
nearly fixed!
one of my pickers does {source} {preset} for the filename, and it still takes the {preset} part from the currently set preset in the main window.

# VidCoder 8.21
# Starting job 1/1
#   Source path: S:\#test p\wcx.22.12.31.rebecca.volpetti.wsg.21.mp4
#   Destination path: S:\#test p\wcx.22.12.31.rebecca.volpetti.wsg.21 **TV 1080p NVE 34 U 2.0**.mp4
#   Title: 1
#   Range: All
#   Preset: 1080p NVE 33 U

![Screenshot 2023-01-17 101141](https://user-images.githubusercontent.com/20132137/212858235-ffa4debc-b36f-4163-93af-015a73fa9333.png)

unfortunately, also this https://github.com/RandomEngy/VidCoder/issues/1095

**Comment by RandomEngy** [comment]
Thanks, I made a fix that {preset} issue that will release with 8.22. As for the remaining issue, a log would be helpful.

**Comment by snottyone** [comment]
thank you VERY much!

**Comment by snottyone** [comment]
thank you VERY much!
